packages/ui-components/src/components/Map/Legend/Legend.js

```diff
@@ -78,11 +78,16 @@ export const Legend = React.memo(
       // measure info for mapOverlayCode may not exist when location changes.
       const baseSerieses = baseMeasureInfo[mapOverlayCode]?.[seriesesKey] || [];
       const serieses = baseSerieses.filter(
-        ({ type, hideFromLegend, values = [] }) =>
-          ![MEASURE_TYPE_RADIUS, MEASURE_TYPE_POPUP_ONLY].includes(type) &&
-          hideFromLegend !== true &&
-          // Spetrum legend has values = []
-          (values.length === 0 || values.filter(value => !value?.hideFromLegend).length > 0),
+        ({ type, hideFromLegend, values = [], min, max, noDataColour }) => {
+          // if type is radius or popup-only, don't create a legend
+          if ([MEASURE_TYPE_RADIUS, MEASURE_TYPE_POPUP_ONLY].includes(type)) return false;
+          // if hideFromLegend is true, don't create a legend
+          if (hideFromLegend === true) return false;
```

Comment:
```suggestion
          if (hideFromLegend) return false;
```

I have a personal vendetta against checking boolean values like this 😤